go/tools/builders/env.go

```diff
@@ -49,6 +49,29 @@ func abs(path string) string {
 	}
 }
 
+func envUpdate(old, new []string) []string {
+	retEnvMap := map[string]string{}
+	retEnv := []string{}
+	for _, e := range old {
+		pair := strings.SplitN(e, "=", 2)
+		k, v := pair[0], pair[1]
+		retEnvMap[k] = v
+	}
+	for _, e := range new {
+		pair := strings.SplitN(e, "=", 2)
+		k, v := pair[0], pair[1]
+		if preV := retEnvMap[k]; preV != "" {
+			retEnvMap[k] = preV + " " + v
```

Comment:
This scares me. Seems like we could end up with `GOOS=linux linux` or something. Why is this needed?

Reply:
this is true. however, both ends (old and new) are controlled by rules_go. But in the case of `CGO_CFLAGS` for instance, we could get to the case when it would get overridden instead of merged

Reply:
by both end I mean the skylark and go builder side